vbfml/scripts/err_analysis

```diff
@@ -455,5 +489,313 @@ def compare_hist(ctx, input_dirs: str, number_epochs: int, save_name: str) -> No
+        data["fpr"][middle_thres_roc_index],
+        data["tpr"][middle_thres_roc_index],
+        "ko",
+        label="score cut 0.5",
+    )
+    plt.title("ROC curve")
+    plt.xlabel("False positive rate")
+    plt.ylabel("True positive rate")
+    plt.legend()
+    plt.savefig(f"{outdir}/roc_curve.pdf")
+
+    plt.figure(2)
+    plt.title("Precision recall curve")
+    plt.xlabel("recall")
+    plt.ylabel("precision")
+    plt.legend()
+    plt.savefig(f"{outdir}/prc_curve.pdf")
+
+    if simple_cut:
+        # plot histograms of the mjj distribution for the
+        features, predictions, _, labels = load_cache(input_dir[0])
+        d = pd.DataFrame(
+            {
+                "mjj": features["mjj"],
+                "detajj": features["detajj"],
+                "labels": labels,
+                "weights": features["weight"],
+                "score": predictions,
+            }
+        )
+
+        histogram(data=d, variable="mjj", cut=mjj_cut, outdir=outdir)
+        histogram(data=d, variable="detajj", cut=eta_cut, outdir=outdir)
+        histogram(data=d, variable="score", cut=0.5, outdir=outdir)
+
+
+@cli.command()
+@click.pass_context
+@click.option(
+    "-i",
+    "--input-dir",
+    required=True,
+    help="path to the prediction directory in a model",
+)
+def plot_correl(ctx, input_dir: str) -> None:
+    """
+    plot the correlation plot for all variable available according to the score.
+    """
+
+    # load data
+    features, predictions, label_encoding, _ = load_cache(input_dir)
+
+    # create outdir
+    outdir = pjoin(input_dir, "correlation_scatter")
+    if not os.path.exists(outdir):
+        os.makedirs(outdir)
+
+    # prepare variables for table
+    correl = []
+
+    markersize = 100 / np.sqrt(len(predictions))
+    count = 1
+    # plot correlation plots
+    for (featurename, feature) in features.iteritems():
+
+        print(f"plot {featurename} ({count}/{len(features.keys())})")
+
+        # take only the positive value of eta to find a linear correlation
+        if (featurename == "leadak4_eta") or (featurename == "trailak4_eta"):
+            feature = abs(feature)
+            featurename = f"abs_val_{featurename}"
+
+        correlation = np.corrcoef(predictions, feature)[0, 1]
+        correl.append([correlation, featurename])
+
+        plt.figure()
+        plt.scatter(
+            predictions, feature, s=markersize, label=f"correlation = {correlation:.3f}"
+        )
+        plt.title(f"score correlation with {featurename}")
+        plt.xlabel("Score")
+        plt.ylabel(featurename)
+        plt.legend()
+        plt.savefig(pjoin(outdir, f"correl_{featurename}.pdf"))
+        count += 1
+
+    def absfirst(a: list):
+        return abs(a[0])
+
+    correl.sort(key=absfirst, reverse=True)
+    correl = np.array(correl)
+
+    table = {"variables": correl[:, 1], "score_correlation": correl[:, 0]}
+    table_print = tabulate(table, headers="keys")
+    print(table_print)
```

Comment:
Can you print some message saying that the table is being saved to `table.txt`, so that user executing the code is aware? Also it might be more clear to rename this file to something like `correlation_table.txt`. Maybe something like:
```
table_outfile = pjoin(outdir, "correlation_table.txt")
print(f"Saving correlation data to: {table_outfile}")
print(table_print)

with open(table_outfile, "w") as f:
     f.write(table_print)

```

Reply:
good idea
I changed it